hazelcast/src/jepsen/hazelcast.clj

```diff
@@ -166,7 +162,83 @@
       (assoc op :type :ok, :value (.incrementAndGet atomic-long)))
 
     (teardown! [this test]
-      (.terminate conn))))
+      (.shutdown conn))))
+
+
+(defn create-cp-atomic-long
+  "Creates a new CP based AtomicLong"
+  [client name]
+  (.getAtomicLong (.getCPSubsystem client) name))
+
+
+(defn create-cp-atomic-reference
+  "Creates a new CP based AtomicReference"
+  [client name]
+  (.getAtomicReference (.getCPSubsystem client) name))
+
+
+(defn cp-atomic-long-id-client
+  "Generates unique IDs using a CP AtomicLong"
+  [conn atomic-long]
+  (reify client/Client
+    (setup! [_ test node]
+      (let [conn (connect node)]
+        (cp-atomic-long-id-client conn (create-cp-atomic-long conn "jepsen.atomic-long"))))
+
+    (invoke! [this test op]
+      (assert (= (:f op) :generate))
+      (assoc op :type :ok, :value (.incrementAndGet atomic-long)))
+
+    (teardown! [this test]
+      (.shutdown conn))))
+
+(defn cp-cas-long-client
+  "A CAS register using a CP AtomicLong"
+  [conn atomic-long]
+  (reify client/Client
+    (setup! [_ test node]
+      (let [conn (connect node)]
+        (cp-cas-long-client conn (create-cp-atomic-long conn "jepsen.cas-long"))))
+
+    (invoke! [this test op]
+      (case (:f op)
+        :read (assoc op :type :ok, :value (.get atomic-long))
+        :write (do (.set atomic-long (:value op))
+                   (assoc op :type :ok))
+        :cas (let [[currentV newV] (:value op)]
+               (if (.compareAndSet atomic-long currentV newV)
+                 (assoc op :type :ok)
+                 (assoc op :type :fail :error :cas-failed)
```

Comment:
This isn't super important, but idomatically, you'd close off your parens at the end of this line, instead of scattering them onto later lines. You might want to do a pass through the source to clean these up; not critical, but it'll help you blend in with normal Clojure style. :)